onnxruntime/core/optimizer/qdq_transformer/selectors_actions/qdq_selectors.cc

```diff
@@ -414,6 +435,87 @@ bool MatMulNodeGroupSelector::Check(const GraphViewer& graph_viewer,
   }
 }
 
+bool DQMatMulNodeGroupSelector::Check(const GraphViewer& graph_viewer,
+                                      const Node& node,
+                                      const std::vector<const Node*>& dq_nodes,
+                                      const std::vector<const Node*>& q_nodes) const {
+  ONNX_UNUSED_PARAMETER(q_nodes);
+  const auto& graph = graph_viewer.GetGraph();
+
+  // MatMul has only 1 DQ input and the DQ must has 1 output edge which is not graph output
+  if (dq_nodes.size() != 1 || !optimizer_utils::CheckOutputEdges(graph, *dq_nodes[0], 1)) {
+    return false;
+  }
+
+  // DQ must be MatMul's the second input
+  if (node.InputDefs()[1]->Name() != dq_nodes[0]->OutputDefs()[0]->Name()) {
```

Comment:
can directly check `NodeArg*` equality
```suggestion
  if (node.InputDefs()[1] != dq_nodes[0]->OutputDefs()[0]) {
```